### PR TITLE
Upgrading to SS4

### PR DESCRIPTION
**Pull request by ScopeyNZ**
As titled.

### CODE REVIEW

**Comment by ScopeyNZ** [comment]
Tests are passing on most platforms but I guess the combination of this module being memory intensive and generating a coverage report is making the 7.2 test fail - out of memory. Any suggestions?

I tried updated travis.yml and enforce higher memory when running coverage but it seems to just completely ignore it.

**Comment by robbieaverill** [comment]
Yep, you can bump the PHP memory limit in Travis: https://github.com/silverstripe/silverstripe-userforms/blob/master/.travis.yml#L22

We had a similar problem in framework yesterday

File: tests/Extensions/ComposerUpdateExtensionTest.php [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 namespace BringYourOwnIdeas\UpdateChecker\Tests\Extensions;
 
+use BringYourOwnIdeas\Maintenance\Model\Package;
 use BringYourOwnIdeas\UpdateChecker\Extensions\ComposerUpdateExtension;
-use Package;
 use PHPUnit_Framework_TestCase;
-use SapphireTest;
```
Comment (robbieaverill) [inline tests/Extensions/ComposerUpdateExtensionTest.php:8]: `protected $requiredExtensions` will need to become `protected static $required_extensions` in this file

File: tests/Extensions/CheckComposerUpdatesExtensionTest.php [hunk 2]
```diff
@@ -36,14 +34,20 @@ public function setUp()
     {
         parent::setUp();
 
+        // Register the extension stub for unit testing to avoid loading Composer
```
Comment (robbieaverill) [inline tests/Extensions/CheckComposerUpdatesExtensionTest.php:37]: `setUp()` should be protected with PHPUnit 5

File: src/UpdateChecker.php [hunk 3]
```diff
@@ -9,9 +9,10 @@
 use Composer\Package\PackageInterface;
 use Composer\Package\Version\VersionSelector;
 use Composer\Repository\CompositeRepository;
-use Package;
-use DataObject;
-use Injector;
+
+
+
+use SilverStripe\Core\Injector\Injector;
```
Comment (robbieaverill) [inline src/UpdateChecker.php]: Can you phpcbf this?

**Review by robbieaverill — CHANGES_REQUESTED** [review]
Looks good over all =)

**Review by robbieaverill — APPROVED** [review]

**Comment by robbieaverill** [comment]
🎉

**Pull request merged**